magicgui/backends/_ipynb/widgets.py

```diff
@@ -0,0 +1,425 @@
+from typing import Any, Callable, Iterable, Optional, Tuple, Type, Union
+
+try:
+    import ipywidgets
+    from ipywidgets import widgets as ipywdg
+except ImportError as e:
+    raise ImportError(
+        "magicgui requires ipywidgets to be installed to use the 'ipynb' backend. "
+        "Please run `pip install ipywidgets`"
+    ) from e
+
+from magicgui.widgets import _protocols
+from magicgui.widgets._bases import Widget
+
+
+def _pxstr2int(pxstr: Union[int, str]) -> int:
+    if isinstance(pxstr, int):
+        return pxstr
+    if isinstance(pxstr, str) and pxstr.endswith("px"):
+        return int(pxstr[:-2])
+    return int(pxstr)
+
+
+def _int2pxstr(pxint: Union[int, str]) -> str:
+    return f"{pxint}px" if isinstance(pxint, int) else pxint
+
+
+class _IPyWidget(_protocols.WidgetProtocol):
+    _ipywidget: ipywdg.Widget
+
+    def __init__(
+        self, qwidg: Type[ipywdg.Widget] = None, parent: Optional[ipywdg.Widget] = None
```

Comment:
I know that this is inheritance, but `qwidg` name is not fitting to context for me. Maybe some update of the name?